src/Signature/SignatureProvider.php

```diff
@@ -109,6 +109,7 @@ public static function version()
     {
         return function ($version, $service, $region) {
             switch ($version) {
+                case 's3v4':
```

Comment:
It might be a good idea to add a test to ensure s3v4 can be resolved. While it's dead simple right now, it would help in case of a future refactor.